server/middleware/setupPdfRenderer.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response, NextFunction } from 'express'
+import GotenbergClient, { PdfOptions } from '../data/gotenbergClient'
+import logger from '../../logger'
+
+/*
+ * This function accepts a Gotenberg client as its only argument.
+ * It returns a handler function to render a normal HTML view template to
+ * produce the raw HTML (including images, stylesheet etc). It then uses a
+ * callback function to pass rendered HTML view into the Gotenberg client
+ * to produce and return a PDF document.
+ */
+
+export default function setupPdfRenderer(client: GotenbergClient) {
+  return (req: Request, res: Response, next: NextFunction) => {
+    res.renderPDF = (
+      view: string,
+      pageData: object,
+      options: { filename: string; pdfOptions?: PdfOptions } = { filename: 'document.pdf' }
+    ): void => {
+      res.render(view, pageData, async (error: Error, html: string) => {
+        if (error) {
+          throw error
+        }
+
+        res.header('Content-Type', 'application/pdf')
+        res.header('Content-Transfer-Encoding', 'binary')
+        res.header('Content-Disposition', `attachment; filename=${options.filename}`)
```

Comment:
`Content-Disposition: attachment; ....` causes the browser to download the file without displaying it
`Content-Disposition: inline; ....` causes the browser to render the file in the browser tab

Maybe we should make this configurable and part of the `options` object

Reply:
Could we make the filename of a similar format to the barcode image download - just think it will help if there are multiple prints in the print queue?

Reply:
Yes, that was my thinking, though not sure what to do if there are multiple barcodes in the 1 PDF (not that thats in scope for this ticket, but we know its coming)
Its one of the questions I have for Liora and Lauren ....